src/Microsoft.Kusto.ServiceLayer/DataSource/KustoClient.cs

```diff
@@ -49,21 +51,23 @@ public KustoClient(string connectionString, string azureAccountToken, string own
 
         private GlobalState LoadSchemaState()
         {
-            CancellationTokenSource source = new CancellationTokenSource();
-
             IEnumerable<ShowDatabaseSchemaResult> tableSchemas = Enumerable.Empty<ShowDatabaseSchemaResult>();
             IEnumerable<ShowFunctionsResult> functionSchemas = Enumerable.Empty<ShowFunctionsResult>();
 
-            Parallel.Invoke(() =>
+            if (!string.IsNullOrWhiteSpace(DatabaseName))
             {
-                tableSchemas = ExecuteControlCommandAsync<ShowDatabaseSchemaResult>(
-                    $".show database {DatabaseName} schema",
-                    false, source.Token).Result;
-            }, () =>
-            {
-                functionSchemas = ExecuteControlCommandAsync<ShowFunctionsResult>(".show functions", false,
-                    source.Token).Result;
-            });
+                var source = new CancellationTokenSource();
```

Comment:
It does not seem right to create a cancellation token here, as the user will not be able to use this to cancel anything. This should come through the call which is initializing state. I think we should use the RetryPolicy which is set up in the ReliableDataSourceConnection. There might be a cancellation token being passed in when this operation is invoked.